test/redis/cache_impl_test.go

```diff
@@ -529,3 +420,107 @@ func TestRedisWithJitter(t *testing.T) {
 	assert.Equal(uint64(0), limits[0].Stats.OverLimit.Value())
 	assert.Equal(uint64(0), limits[0].Stats.NearLimit.Value())
 }
+
+func mustNewRedisServer() *miniredis.Miniredis {
+	srv, err := miniredis.Run()
+	if err != nil {
+		panic(err)
+	}
+
+	return srv
+}
+
+func TestNewClientImpl(t *testing.T) {
+	redisAuth := "123"
+	statsStore := stats.NewStore(stats.NewNullSink(), false)
+
+	mkRedisClient := func(auth, addr string) redis.Client {
+		return redis.NewClientImpl(statsStore, false, auth, addr, 1, 1*time.Millisecond, 1)
+	}
+
+	t.Run("no connection", func(t *testing.T) {
+		assert.Panics(t, func() {
+			// It's possible there is a redis server listening on 6379 in ci environment, so
+			// use a random port.
+			mkRedisClient("", "localhost:12345")
```

Comment:
Is it to test the [ping/pong connection](https://github.com/envoyproxy/ratelimit/blob/d08853c479faffb87b799e28baf4580e9159eb48/src/redis/driver_impl.go#L87-L90) ? can we also check the error?

Reply:
It is to test connection refused case. `ping/pong` is used to check auth.(we don't know if auth is ok util receive first response from redis).
Sure we can check the error. It require update testify package.